server/controllers/availability.js

```diff
@@ -0,0 +1,124 @@
+const Availability = require("../models/Availability");
+const asyncHandler = require("express-async-handler");
+const mongoose = require("mongoose");
+
+// @route POST /availability
+// @desc create a schedule
+// @access private
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+    const { schedule } = req.body.schedule
+    const profile = await Profile.findOne({userId: req.user.id});
+
+    if(!profile){
+      res.status(404)
+      throw new Error("There is no profile associated with this user!")
+    } else {
+      const newSchedule = new Availability({
+        profileId: profile._id,
+        schedule: schedule
+      })
+  
+      const savedSchedule = await newSchedule.save()
+  
+      if(!savedSchedule){
+        res.status(500)
+        throw new Error("Schedule could not be created!")
+      } else {
+        res.status(201).send(savedSchedule)
+      }
+    }
+  });
+
+  // @route GET /availability/:scheduleId
+  // @desc get a specific schedule from a profile 
+  // @access public
+  exports.getSchedule = asyncHandler(async (req, res, next) => {
+    const schedule = await Availability.findOne({ _id: req.params.scheduleId})
+    const profile = await Profile.findOne({userId: req.user.id});
+
+    if(schedule.profileId === profile._id) {
+      if(!schedule){
```

Comment:
```suggestion
      if(!schedule){
```
Let's move this nested `if` clause above lin 37